src/main/java/com/coveo/spillway/storage/AsyncLimitUsageStorage.java

```diff
@@ -78,6 +78,14 @@ public AsyncLimitUsageStorage(LimitUsageStorage wrappedLimitUsageStorage) {
     return wrappedLimitUsageStorage.debugCurrentLimitCounters();
   }
 
+  public void shutdownStorage() {
+    executorService.shutdown();
```

Comment:
Ho I see what you mean. Well the shutdown isn't meant to be called during the life of the async storage. Its something that I would have put in a destructor, but Java doesn't have one...